app/src/test/kotlin/fr/free/nrw/commons/media/MediaDetailFragmentUnitTests.kt

```diff
@@ -229,6 +231,22 @@ class MediaDetailFragmentUnitTests {
         fragment.launchZoomActivity(view)
     }
 
+    @Test
+    @Throws(Exception::class)
+    fun testOnUpdateCoordinatesClicked() {
+        `when`(media.coordinates).thenReturn(null)
+        `when`(applicationKvStore.getString(LAST_LOCATION)).thenReturn("37.773972,-122.431297")
+        fragment.onUpdateCoordinatesClicked()
```

Comment:
Since `media` is a mock you can verify if `media.getCoordinates()` is invoked 3 times or not.

You can Assert for Shadow Activity Intent and check if LocationPickerActivity is launched or not.

Reply:
how can i check if media.getCoordinates is invoked 3 times?

Reply:
You can simply use `verify(media, times(3)).getCoordinates() `